[UPDATE MAIN GUIDE] Update with fixing signature check Bitcoin Core + dependencies update (#1238)
raspibolt/raspibolt: guide/bitcoin/bitcoin-client.md

### Signature check

> Bitcoin releases are signed by several individuals, each using its own key. To verify the validity of these signatures, you must first import the corresponding public keys into your GPG key database.

Reviewer: ```suggestion
Bitcoin releases are signed by several individuals, each using their own key. To verify the validity of these signatures, you must first import the corresponding public keys into your GPG key database.
```

* Create ".gnupg" folder by typing the next command

  ```sh
  $ gpg --list-keys
  ```

* Enter on the ".gnupg" folder, create a persistent folder for the signatures called "sigs" and enter on it

  ```sh
  $ cd /home/admin/.gnupg/ && mkdir sigs && cd sigs
  ```

* The next command download and imports automatically all signatures from the [Bitcoin Core release attestations (Guix)](https://github.com/bitcoin-core/guix.sigs) repository

  ```sh

>   $ curl 'https://api.github.com/repositories/355107265/contents/builder-keys' | jq '.[] .download_url' | xargs -L1 wget -N && curl 'https://api.github.com/repositories/355107265/contents/builder-keys' | jq '.[] .name' | xargs -L1 gpg --import

Author: If you have a better solution, tell us

>   $ curl 'https://api.github.com/repositories/355107265/contents/builder-keys' | jq '.[] .download_url' | xargs -L1 wget -N && curl 'https://api.github.com/repositories/355107265/contents/builder-keys' | jq '.[] .name' | xargs -L1 gpg --import

Reviewer 2: This works without `jq` and do one download less.
```suggestion
  $ curl -s "https://api.github.com/repositories/355107265/contents/builder-keys" | grep download_url | grep -oE "https://[a-zA-Z0-9./-]+" | while read url; do curl -s "$url" | gpg --import; done
```

Reviewer: If we go ahead with this change let's update "For the future: upgrade Bitcoin Core" as well and remove the `Command 'jq' not found` item below. Also can then revert any changes with ssystem-configuration.md.

  ```

Expected output:

  ```sh
  >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
  >                                 Dload  Upload   Total   Spent    Left  Speed
  > 100 30520  100 30520    0     0   200k      0 --:--:-- --:--:-- --:--:--  198k
  > gpg: key 17565732E08E5E41: 29 signatures not checked due to missing keys
  > gpg: /home/admin/.gnupg/trustdb.gpg: trustdb created
  > gpg: key 17565732E08E5E41: public key "Andrew Chow <andrew@achow101.com>" imported
  > gpg: Total number processed: 1
  > gpg:               imported: 1
  > gpg: no ultimately trusted keys found
  [...]
  ```

> 💡 If appears to you `Command 'jq' not found` you will need to install `jq` with `$ sudo apt install jq`. When it finishes, repeat the step before.

Reviewer: ```suggestion
💡 If, instead of the expected output above, you see `Command 'jq' not found` you will need to install `jq` by executing the following command `$ sudo apt install jq`. After which, repeat previous cURL command and verify the results.
```

> * Return to the `tmp` folder and follow the signature check process

Reviewer: ```suggestion
* Return to the `tmp` folder and complete the signature check process
```

  $ wget https://bitcoincore.org/bin/bitcoin-core-24.0.1/SHA256SUMS.ots
  ```

* Verify the new version against its checksums

  ```sh
  $ sha256sum --ignore-missing --check SHA256SUMS
  > bitcoin-24.0.1-aarch64-linux-gnu.tar.gz: OK
  ```

* The next command download and imports automatically all signatures from the [Bitcoin Core release attestations (Guix)](https://github.com/bitcoin-core/guix.sigs) repository

  ```sh
  $ curl -s "https://api.github.com/repositories/355107265/contents/builder-keys" | grep download_url | grep -oE "https://[a-zA-Z0-9./-]+" | while read url; do curl -s "$url" | gpg --import; done
  ```

Expected output:

  ```sh
  >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
  >                                 Dload  Upload   Total   Spent    Left  Speed

>   > 100 30520  100 30520    0     0   200k      0 --:--:-- --:--:-- --:--:--  198k

Reviewer 2: There will be no this part, as updated commands use `curl -s`.

Author: Fixed!

  > gpg: key 17565732E08E5E41: 29 signatures not checked due to missing keys
  > gpg: /home/admin/.gnupg/trustdb.gpg: trustdb created
  > gpg: key 17565732E08E5E41: public key "Andrew Chow <andrew@achow101.com>" imported
  > gpg: Total number processed: 1
  > gpg:               imported: 1
  > gpg: no ultimately trusted keys found
  [...]
  ```

* Verify that the checksums file is cryptographically signed by the release signing keys.
  The following command prints signature checks for each of the public keys that signed the checksums.

  ```sh
  $ gpg --verify SHA256SUMS.asc
  ```

* Check that at least a few signatures show the following text

  ```sh
  > gpg: Good signature from ...
  > Primary key fingerprint: ...
  ```

> * Verify the timestamp. If the prompt shows you `-bash: ots: command not found`, ensure that you are installing correctly OTS client in the [proper section](#opentimestamps-client)

Reviewer: ```suggestion
* Verify the timestamp. If the prompt shows you `-bash: ots: command not found`, ensure that you are installing correctly OTS client in the [proper section](bitcoin-client.md#opentimestamps-client)
```

  $ wget https://bitcoincore.org/bin/bitcoin-core-24.0.1/SHA256SUMS.ots
  ```

* Verify the new version against its checksums

  ```sh
  $ sha256sum --ignore-missing --check SHA256SUMS
  > bitcoin-24.0.1-aarch64-linux-gnu.tar.gz: OK
  ```

* Enter the `sigs` folder

  ```sh
  $ cd /home/admin/.gnupg/sigs
  ```

* The next command download and imports automatically all signatures from the Bitcoin Core Guix repository and update our database if necessary

  ```sh

>   $ curl 'https://api.github.com/repositories/355107265/contents/builder-keys' | jq '.[] .download_url' | xargs -L1 wget -N && curl 'https://api.github.com/repositories/355107265/contents/builder-keys' | jq '.[] .name' | xargs -L1 gpg --import

Reviewer: Is it worth adding the note again here in case of an error (Command 'jq' not found)? Since the previous section where it was mentioned would only be applicable to new installations. Anyone upgrading having not followed this initial process may well run into the issue and be confused. Alternatively, you could always hyperlink to that earlier section.

Author: `💡 If appears to you `Command 'jq' not found` you will need to install `jq` with `$ sudo apt install jq`. When it finishes, repeat the step before.`

I think one time is enough

Reviewer: OK no problem. Happy with this for now, not difficult to find the info if someone gets the problem during the next Bitcoin Client update.
